src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -393,6 +395,34 @@ public ClusterState sendClusterStateRequest(TransportService transportService) {
         return clusterStateResponseHandler.getClusterState();
     }
 
+    /**
+     * Request the Dependency Information from Opensearch.
+     * @param transportService
+     * @return The dependency information of Extension
+     */
+
+    public List<DiscoveryExtensionNode> sendExtensionDependencyRequest(TransportService transportService) {
+        logger.info("Sending Extension Dependency Information request to Opensearch");
+        ExtensionDependencyResponseHandler extensionDependencyResponseHandler = new ExtensionDependencyResponseHandler();
+        try {
+            transportService.sendRequest(
+                opensearchNode,
+                ExtensionsManager.REQUEST_EXTENSION_DEPENDENCY_INFORMATION,
+                new ExtensionRequest(ExtensionsManager.RequestType.REQUEST_EXTENSION_DEPENDENCY_INFORMATION),
+                extensionDependencyResponseHandler
+            );
+            // Wait on Extension Dependency response
+            extensionDependencyResponseHandler.awaitResponse();
+        } catch (InterruptedException e) {
+            logger.info("Failed to receive Extension Dependency response from OpenSearch", e);
+        } catch (Exception e) {
```

Comment:
if we're doing the same message either way, we should just have a single catch block.  Should also probably be a WARN level.